src/legacy/core_plugins/data/public/search/aggs/agg_config.test.ts

```diff
@@ -0,0 +1,499 @@
+            params: {},
+          },
+        ],
+      },
+    ];
+
+    testsDifferent.forEach((test, index) => {
+      it(`different aggregations (${index})`, () => {
+        const ac1 = new AggConfigs(indexPattern, test.config1, { typesRegistry });
+        const ac2 = new AggConfigs(indexPattern, test.config2, { typesRegistry });
+        expect(ac1.jsonDataEquals(ac2.aggs)).toBe(false);
+      });
+    });
+  });
+
+  describe('#toJSON', () => {
+    it('includes the aggs id, params, type and schema', () => {
+      const ac = new AggConfigs(indexPattern, [], { typesRegistry });
+      const configStates = {
+        enabled: true,
+        type: 'date_histogram',
+        schema: 'segment',
+        params: {},
+      };
+      const aggConfig = ac.createAggConfig(configStates);
+
+      expect(aggConfig.id).toBe('1');
+      expect(typeof aggConfig.params).toBe('object');
+      expect(aggConfig.type).toBeInstanceOf(AggType);
+      expect(aggConfig.type).toHaveProperty('name', 'date_histogram');
+      expect(typeof aggConfig.schema).toBe('object');
+      expect(aggConfig.schema).toHaveProperty('name', 'segment');
+
+      const state = aggConfig.toJSON();
+      expect(state).toHaveProperty('id', '1');
+      expect(typeof state.params).toBe('object');
+      expect(state).toHaveProperty('type', 'date_histogram');
+      expect(state).toHaveProperty('schema', 'segment');
+    });
+
+    it('test serialization  order is identical (for visual consistency)', () => {
+      const configStates = [
+        {
+          enabled: true,
+          type: 'date_histogram',
+          schema: 'segment',
+          params: {},
+        },
+      ];
+      const ac1 = new AggConfigs(indexPattern, configStates, { typesRegistry });
+      const ac2 = new AggConfigs(indexPattern, configStates, { typesRegistry });
+
+      // this relies on the assumption that js-engines consistently loop over properties in insertion order.
+      // most likely the case, but strictly speaking not guaranteed by the JS and JSON specifications.
+      expect(JSON.stringify(ac1.aggs) === JSON.stringify(ac2.aggs)).toBe(true);
+    });
+  });
+
+  describe('#makeLabel', () => {
+    let aggConfig: AggConfig;
+
+    beforeEach(() => {
+      const ac = new AggConfigs(indexPattern, [], { typesRegistry });
+      aggConfig = ac.createAggConfig({ type: 'count' } as CreateAggConfigParams);
+    });
+
+    it('uses the custom label if it is defined', () => {
+      aggConfig.params.customLabel = 'Custom label';
+      const label = aggConfig.makeLabel();
+      expect(label).toBe(aggConfig.params.customLabel);
+    });
+
+    it('default label should be "Count"', () => {
+      const label = aggConfig.makeLabel();
+      expect(label).toBe('Count');
+    });
+
+    it('default label should be "Percentage of Count" when percentageMode is set to true', () => {
+      const label = aggConfig.makeLabel(true);
+      expect(label).toBe('Percentage of Count');
+    });
```

Comment:
@ppisljar Similar question about `Vis` adding `count` affects this section too. In the original browser tests we had:
```js
      const vis = new visualizationsStart.Vis(indexPattern, {});
      const aggConfig = vis.aggs.aggs[0];
      const label = aggConfig.makeLabel();
      expect(label).to.be('Count');
```

So an empty visState will still create a `count` agg. But here (line 381) I had to explicitly add the type in order for tests to pass:
```ts
aggConfig = ac.createAggConfig({ type: 'count' } as CreateAggConfigParams);
```

Want to double-check this is expected.

Reply:
this is expected, you can achieve the same test by passing schemas as the third argument when creating AggConfigs